src/h3lib/lib/localij.c

```diff
@@ -702,12 +702,10 @@ H3Error H3_EXPORT(gridPathCells)(H3Index start, H3Index end, H3Index *out) {
     ijkToCube(&startIjk);
     ijkToCube(&endIjk);
 
-    double iStep =
-        distance ? (double)(endIjk.i - startIjk.i) / (double)distance : 0;
-    double jStep =
-        distance ? (double)(endIjk.j - startIjk.j) / (double)distance : 0;
-    double kStep =
-        distance ? (double)(endIjk.k - startIjk.k) / (double)distance : 0;
+    double invDistance = distance ? 1.0 / (double)distance : 0;
+    double iStep = distance ? (double)(endIjk.i - startIjk.i) * invDistance : 0;
+    double jStep = distance ? (double)(endIjk.j - startIjk.j) * invDistance : 0;
+    double kStep = distance ? (double)(endIjk.k - startIjk.k) * invDistance : 0;
```

Comment:
Since `invDistance` will be `0` if `distance == 0`, it looks like we could further simplify this to drop the ternaries?
```suggestion
    double invDistance = distance ? 1.0 / (double)distance : 0;
    double iStep = (double)(endIjk.i - startIjk.i) * invDistance;
    double jStep = (double)(endIjk.j - startIjk.j) * invDistance;
    double kStep = (double)(endIjk.k - startIjk.k) * invDistance;
```